source/uwp/Renderer/lib/XamlHelpers.cpp

```diff
@@ -644,84 +644,71 @@ namespace AdaptiveNamespace::XamlHelpers
-
-        // Create an inline for the *
-        ComPtr<IRun> starRun = XamlHelpers::CreateXamlClass<IRun>(HStringReference(RuntimeClass_Windows_UI_Xaml_Documents_Run));
-
-        HString starString;
-        UTF8ToHString(" *", starString.GetAddressOf());
+        ABI::AdaptiveNamespace::ForegroundColor textColor;
+        RETURN_IF_FAILED(inputLabelConfig->get_Color(&textColor));
+        // If we're formatting a hint and the color has been set to default, then we have to render the button with
+        // attention color, otherwise match the label color
+        if (isHint && textColor == ABI::AdaptiveNamespace::ForegroundColor::Default)
+        {
+            textColor = ABI::AdaptiveNamespace::ForegroundColor::Attention;
+        }
 
-        RETURN_IF_FAILED(starRun->put_Text(starString.Get()));
+        ABI::Windows::UI::Color color;
+        RETURN_IF_FAILED(GetColorFromAdaptiveColor(hostConfig, textColor, ContainerStyle_Default, false, false, &color));
 
-        // Set the color to Attention color
-        ComPtr<IAdaptiveHostConfig> hostConfig;
-        RETURN_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
+        ComPtr<ABI::Windows::UI::Xaml::Documents::IRun> xamlLabelRun(labelRun);
 
-        ABI::Windows::UI::Color attentionColor;
-        RETURN_IF_FAILED(
-            GetColorFromAdaptiveColor(hostConfig.Get(), ForegroundColor_Attention, ContainerStyle_Default, false, false, &attentionColor));
+        ComPtr<ABI::Windows::UI::Xaml::Documents::ITextElement> labelRunAsTextElement;
+        RETURN_IF_FAILED(xamlLabelRun.As(&labelRunAsTextElement));
 
-        ComPtr<ABI::Windows::UI::Xaml::Documents::ITextElement> starRunAsTextElement;
-        RETURN_IF_FAILED(starRun.As(&starRunAsTextElement));
+        RETURN_IF_FAILED(labelRunAsTextElement->put_Foreground(XamlHelpers::GetSolidColorBrush(color).Get()));
 
-        RETURN_IF_FAILED(starRunAsTextElement->put_Foreground(XamlHelpers::GetSolidColorBrush(attentionColor).Get()));
+        ABI::AdaptiveNamespace::TextSize textSize;
+        RETURN_IF_FAILED(inputLabelConfig->get_Size(&textSize));
 
-        ComPtr<ABI::Windows::UI::Xaml::Documents::IInline> starRunAsInline;
-        RETURN_IF_FAILED(starRun.As(&starRunAsInline));
+        UINT32 resultSize{};
+        RETURN_IF_FAILED(GetFontSizeFromFontType(hostConfig, ABI::AdaptiveNamespace::FontType_Default, textSize, &resultSize));
 
-        RETURN_IF_FAILED(xamlInlines->Append(starRunAsInline.Get()));
+        RETURN_IF_FAILED(labelRunAsTextElement->put_FontSize(resultSize));
 
         return S_OK;
     }
 
-    HRESULT FormatLabelWithHostConfig(_In_ ABI::AdaptiveNamespace::IAdaptiveRenderContext* renderContext,
-                                      _In_ boolean isRequired,
-                                      ABI::Windows::UI::Xaml::Documents::IRun* labelRun)
+    // TextBlock and RichTextBlock xaml types both support inlines, so we reuse the same method for rendering the " *" hint in both cases
```

Comment:
>// TextBlock and RichTextBlock xaml types both support inlines, so we reuse the same method for rendering the " *" hint in both cases [](start = 4, length = 133)

This comment seems obsolete now that we're not supporting rich text blocks as labels, right?